packages/odf/hooks/useNodesData.ts

```diff
@@ -46,16 +49,18 @@ export const useNodesData = (
             )
         )
         .map((node: Partial<NodeData>): NodeData => {
-          const metric = _.find(utilization.data.result, [
-            'metric.instance',
-            node.metadata.name,
-          ]);
+          const metric = utilization?.data?.result
+            ? _.find(utilization?.data?.result, [
+                'metric.instance',
+                getName(node),
+              ])
+            : undefined;
```

Comment:
nits: `metric` will automatically be undefined, if `utilization?.data?.result` is empty.
```suggestion
          const metric = _.find(utilization?.data?.result || [], [
                'metric.instance',
                getName(node),
              ]);
```